### ISSUE TITLE
Investigate build time configuration for `SSLContextParameters`

### ISSUE THREAD

**Issue by jamesnetherton** [apache/camel-quarkus#3094]
We could investigate having some build time configuration for [`SSLContextParameters`](https://camel.apache.org/manual/latest/camel-configuration-utilities.html) which we can use to set up the global SSL config on the `CamelContext`. E.g like `camelContext.setSSLContextParameters(params)`.

Or taken a step further, leverage such configuration to configure Quarkus SSL. E.g for folks coming from a pure Camel world who are used to doing the security setup with `SSLContextParameters`, it could make more sense to them rather than setting the individual Quarkus SSL config properties.

**Comment by ppalaga** [comment]
I wonder whether build time SSLContextParameters would mean that changing a server certificate would require an application rebuild?

**Comment by jamesnetherton** [comment]
> I wonder whether build time SSLContextParameters would mean that changing a server certificate would require an application rebuild?

Yeah that's quite possible. I have no idea if this idea is viable, so the first step is just to investigate a bit.
